incubator/hnc/internal/validators/object_test.go

```diff
@@ -563,6 +590,54 @@ func TestUserChanges(t *testing.T) {
 	}
 }
 
+type fakeNSClient struct {
+	isDeleting bool
+}
+
+// Get decodes given runtime.Object as corev1.Namespace that might contains deletionTimestamp
+func (c fakeNSClient) Get(_ context.Context, key client.ObjectKey, obj client.Object) error {
+	ns := &corev1.Namespace{}
+	if c.isDeleting {
+		ns.SetDeletionTimestamp(&metav1.Time{Time: time.Now()})
+	}
+
+	j, err := json.Marshal(ns)
+	if err != nil {
+		return err
+	}
+	decoder := scheme.Codecs.UniversalDecoder()
+	_, _, err = decoder.Decode(j, nil, obj)
```

Comment:
What's the purpose of having this? The test never looks at anything other than the deletion timestamp, right?

Reply:
Yes, and it will decode return value (with deletionTimestamp) into input pointer `obj`, which in function `Get()` is an interface. This is used to create `fakeClient` in controller-runtime https://github.com/kubernetes-sigs/controller-runtime/blob/v0.8.1/pkg/client/fake/client.go#L275-L281

Reply:
Given that we know that `obj` is actually a pointer to a namespace object in all cases that we care about, can we just cast it?

```
ns := obj.(*corev1.Namespace)
ns.DeletionTimestamp = <whatever>
```